tools/schema_loader.cc

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2021-present ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+#include <boost/algorithm/string.hpp>
+#include <seastar/core/fstream.hh>
+#include <seastar/util/closeable.hh>
+
+#include "cql3/query_processor.hh"
+#include "cql3/statements/create_keyspace_statement.hh"
+#include "cql3/statements/create_table_statement.hh"
+#include "cql3/statements/create_type_statement.hh"
+#include "db/config.hh"
+#include "database.hh"
+#include "gms/feature_service.hh"
+#include "locator/token_metadata.hh"
+#include "locator/snitch_base.hh"
+#include "tools/schema_loader.hh"
+#include "utils/fb_utilities.hh"
+
+namespace {
+
+sstring read_file(std::filesystem::path path) {
+    auto file = open_file_dma(path.string(), open_flags::ro).get();
+    auto fstream = make_file_input_stream(file);
+
+    size_t size = 0;
+    std::vector<temporary_buffer<char>> bufs;
+    while (auto buf = fstream.read().get()) {
+        size += buf.size();
+        bufs.emplace_back(std::move(buf));
+    }
+
+    sstring content(sstring::initialized_later{}, size);
+    char* it = content.begin();
+    for (const auto& buf : bufs) {
+        it = std::copy_n(buf.begin(), buf.size(), it);
+    }
+    return content;
+}
+
+std::vector<sstring> split_statements(sstring statements) {
+    std::vector<sstring> split_res;
+    boost::split(split_res, statements, [] (char c) { return c == ';'; });
+    std::vector<sstring> res;
+    for (auto& stmt : split_res) {
+        if (!std::all_of(stmt.begin(), stmt.end(), [] (const char& c) { return std::isspace(c); })) {
+            res.emplace_back(std::move(stmt));
+        }
+    }
+    return res;
+}
+
+schema_ptr do_load_schema(std::filesystem::path path) {
+    cql3::cql_stats cql_stats;
+
+    db::config cfg;
+    cfg.enable_cache(false);
+    cfg.volatile_system_keyspace_for_testing(true);
```

Comment:
I wonder if instead of having all of these weird setup stuff in yet another source file, we could reuse the "test environment" code that we already have? Or maybe this "weird setup" is just weird in a different way from the test setup? :-)

Reply:
The test environment code does way too much stuff that I don't need, and it also sets up a complete environment with system tables and all. What I really need here is just a `database` instance to prepare the statement, and sadly we need this much stuff to get that working.

Reply:
Also the `cql_test_env` takes many seconds to start and spills a huge amount of logs, something I don't want my tool to do every time it loads a schema from disk.